### PR TITLE
split booking flow to registration and finalization steps

### PR DESCRIPTION
**Pull request by DFSko**
resolves happy-travel/agent-app-project#813

### CODE REVIEW

File: HappyTravel.Edo.DirectApi/Controllers/BookingController.cs [hunk 1]
```diff
@@ -37,17 +37,33 @@ public BookingController(IAgentContextService agentContextService, BookingCancel
...
-                ? Ok(response)
+                ? Ok(booking)
+                : BadRequest(ProblemDetailsBuilder.Build(error));
+        }
+
+
+        /// <summary>
+        /// Finalize booking
+        /// </summary>>
+        [HttpPost("finalize")]
```
Comment (mingazhev) [inline HappyTravel.Edo.DirectApi/Controllers/BookingController.cs:58]: May be using a post method with empty content and our reference code in the route?

File: HappyTravel.Edo.DirectApi/Models/BookingFinalizationRequest.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    public readonly struct BookingFinalizationRequest
+    {
+        public BookingFinalizationRequest(string referenceCode, string supplierReferenceCode)
+        {
+            ReferenceCode = referenceCode;
+            SupplierReferenceCode = supplierReferenceCode;
+        }
+
+
+        public string ReferenceCode { get; }
```
Comment (mingazhev) [inline HappyTravel.Edo.DirectApi/Models/BookingFinalizationRequest.cs:12]: Don't we return this during the booking registration?

File: HappyTravel.Edo.DirectApi/Controllers/BookingController.cs [hunk 3]
```diff
@@ -37,17 +37,33 @@ public BookingController(IAgentContextService agentContextService, BookingCancel
         
         
         /// <summary>
-        /// Creating booking.
+        /// Register booking.
         /// </summary>
         [HttpPost]
-        public async Task<ActionResult<AccommodationBookingInfo>> Book([FromBody] AccommodationBookingRequest request)
+        public async Task<ActionResult<Booking>> Register([FromBody] AccommodationBookingRequest request)
```
Comment (mingazhev) [inline HappyTravel.Edo.DirectApi/Controllers/BookingController.cs:43]: Do we return a whole database entity from controller?
This is unacceptable

Comment (DFSko) [reply HappyTravel.Edo.DirectApi/Controllers/BookingController.cs:43]: It's not database entity, it's direct api dto

File: HappyTravel.Edo.DirectApi/Models/BookingFinalizationRequest.cs (hunk 2 above)
Comment (DFSko) [reply HappyTravel.Edo.DirectApi/Models/BookingFinalizationRequest.cs:12]: We return both reference codes

File: HappyTravel.Edo.DirectApi/Controllers/BookingController.cs (hunk 1 above)
Comment (DFSko) [reply HappyTravel.Edo.DirectApi/Controllers/BookingController.cs]: Fixed

**Review by mingazhev — APPROVED** [review]

**Pull request merged**